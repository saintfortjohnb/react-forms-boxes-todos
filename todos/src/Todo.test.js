import React from 'react';
import { render } from '@testing-library/react';
import Todo from './Todo';

// Smoke Test
it('Todo renders without crashing', () => {
  const mockRemoveTodo = jest.fn();
  render(<Todo id="123" task="Test task" removeTodo={mockRemoveTodo} />);
});

// Snapshot Test
it('Todo matches snapshot', () => {
  const mockRemoveTodo = jest.fn();
  const { asFragment } = render(<Todo id="123" task="Test task" removeTodo={mockRemoveTodo} />);
  expect(asFragment()).toMatchSnapshot();
});
