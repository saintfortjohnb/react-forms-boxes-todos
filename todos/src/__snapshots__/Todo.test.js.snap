// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <span>
      Test task
    </span>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
